.github/workflows/deploy.yml

```diff
@@ -0,0 +1,59 @@
+name: Deploy primodium interface
+
+on:
+  workflow_dispatch:
+  pull_request:
+
+jobs:
+  deploy:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      packages: write
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4.2.0
+
+      - uses: pnpm/action-setup@v4
+
+      - name: Install Foundry
+        uses: foundry-rs/foundry-toolchain@v1
+
+      - name: Build code
+        run: |
+          cat > .env <<-EOF
+          ${{vars.PRIMODIUM_ENV_FILE}}
```

Comment:
we might move to just declaring these inline, as its a bit clearer, but probably fine for now. 
More generally, we should do a refactor of all/these workflows at some stage, to avoid common duplicated code as per https://docs.github.com/en/actions/sharing-automations/avoiding-duplication , Porto do this nicely in their repo: https://github.com/ithacaxyz/porto/tree/main/.github

Reply:
resolved in https://github.com/HappyChainDevs/primodium/pull/8